jolt-core/src/jolt/vm/state_manager.rs

```diff
@@ -0,0 +1,120 @@
+use std::collections::HashMap;
+use std::ops::Index;
+use std::sync::{Arc, Mutex};
+
+use tracer::instruction::RV32IMCycle;
+
+use crate::field::JoltField;
+use crate::poly::commitment::commitment_scheme::CommitmentScheme;
+use crate::poly::opening_proof::{ProverOpeningAccumulator, VerifierOpeningAccumulator};
+use crate::r1cs::inputs::JoltR1CSInputs;
+use crate::utils::transcript::Transcript;
+
+pub type Openings<F> = HashMap<OpeningsKeys, (Vec<F>, F)>;
+
+impl<F> Index<JoltR1CSInputs> for Openings<F> {
+    type Output = F;
+
+    fn index(&self, index: JoltR1CSInputs) -> &Self::Output {
+        &self[&OpeningsKeys::SpartanZ(index)].1
+    }
+}
+
+pub struct StateManager<'a, F: JoltField, PCS: CommitmentScheme<T, Field = F>, T: Transcript> {
+    pub transcript: &'a mut T,
+    pub prover_accumulator: Option<Arc<Mutex<&'a mut ProverOpeningAccumulator<F, PCS, T>>>>,
+    pub verifier_accumulator: Option<Arc<Mutex<&'a mut VerifierOpeningAccumulator<F, PCS, T>>>>,
+    pub openings: Arc<Mutex<Openings<F>>>,
+    pub r_address: Vec<F>,
+}
+
+#[derive(Hash, PartialEq, Eq, Copy, Clone, Debug)]
+pub enum OpeningsKeys {
+    SpartanZ(JoltR1CSInputs),
+    InstructionTypeFlag(usize),
```

Comment:
```suggestion
    LookupTableFlag(LookupTables<32>),
```

Reply:
The `LookupTables` enum doesn't contain just variants, but the structs in those variants as well. But I renamed the enum type.